src/applications/disability-benefits/4142/config/form.js

```diff
@@ -0,0 +1,252 @@
+// _ from 'lodash/fp';
+
+// Example of an imported schema:
+// import fullSchema from '../22-4142-schema.json';
+// In a real app this would be imported from `vets-json-schema`:
+// import fullSchema from 'vets-json-schema/dist/22-4142-schema.json';
+
+// In a real app this would not be imported directly; instead the schema you
+// imported above would import and use these common definitions:
+import commonDefinitions from 'vets-json-schema/dist/definitions.json';
+import PrivateProviderTreatmentView from '../../../../platform/forms/components/ServicePeriodView';
+import dateRangeUI from 'us-forms-system/lib/js/definitions/dateRange';
+
+import IntroductionPage from '../containers/IntroductionPage';
+import ConfirmationPage from '../containers/ConfirmationPage';
+
+// const { } = fullSchema.properties;
+
+// const { } = fullSchema.definitions;
+
+import {
+  medicalRecDescription,
+  letUsKnow,
+  aboutPrivateMedicalRecs,
+  limitedConsentDescription,
+  summary,
+} from '../helpers';
+
+
+import PhoneNumberWidget from 'us-forms-system/lib/js/widgets/PhoneNumberWidget';
+
+const { fullName, ssn, date, dateRange, usaPhone } = commonDefinitions;
+
+// Define all the fields in the form to aid reuse
+const formFields = {
```

Comment:
What problem does defining all the fields in an object like this solve?

Specifically, what I'm thinking about is what happens if a field name is misspelled. Without the reference object like this, we end up with something like:
```js
uiSchema: {
  fulllName: { ... },
  sssn: { ... }
}
```
If these were misspelled object properties, it'll end up like:
```js
uiSchema: {
  undefined: { ... },
  undefined: { ... }
}
```
The second `undefined` will override the first and we'll lose a property. This may or may not break things silently.

Neither are great, but which one is easier to debug?

I've definitely tried to solve this problem before, but I still don't have a good solution, so I'm wide open to suggestions. A month or two ago, I tried using `Proxy` to define a getter method which threw an error if the property accessed wasn't in the object, but IE doesn't support Proxies, so...that was no good. :disappointed:

Reply:
@cvalarida This is an idea that I introduced to the form created by the generator. The basic idea is I was turned off by the reuse of magic strings for each form field. At the very least each form field name is repeated twice, once in the `schema.properties` and once in the `uiSchema`, and sometimes the . So the idea is declare the form name up here in the `formFields` object and actually use them in the `schema` and `uiSchema`: `formFields['firstName']: ...` I'm wide open to other thoughts and ideas!

Reply:
I'm not 100%, but I also think that the form fields need to be unique across all form pages. So defining them here helps encourage making them unique (but does not actually enforce uniqueness because there's nothing to ensure the `formFields` object's *values* are unique)

Reply:
There are certainly weaknesses to my approach. On the plus side, you at least get editor autocomplete (yay!) when accessing the properties on `formFields` in the form config. On the downside, because JS is so loose there are no complaints when you try to access an undefined property on `formFields` :( I guess that's where a library enforcing types would be helpful.

Another plus side to doing something, *anything*, beyond reusing the same string all over the place is that is makes it clear that the `firstName` property on `uiSchema` is the same as the `firstName` on `schema.properties` which is the same as `firstName` in the list of required fields, which is the same as the `firstName` in `expandUnder`. That's something that was not clear to me when I was brand new to this code.

Summary: this method I'm proposing sucks, but I think it sucks less than the current way. One step forward, two steps sideways!